tedana/viz.py

```diff
@@ -5,15 +5,45 @@
 import os
 
 import numpy as np
+import matplotlib
+matplotlib.use('AGG')
 import matplotlib.pyplot as plt
 
 from tedana import model
+from tedana.utils import get_spectrum
 
 LGR = logging.getLogger(__name__)
+MPL_LGR = logging.getLogger('matplotlib')
+MPL_LGR.setLevel(logging.WARNING)
+
+
+def trim_edge_zeros(arr):
+    mask = arr!=0
+    bounding_box = tuple(
+                         slice(np.min(indexes), np.max(indexes) + 1)
+                         for indexes in np.where(mask))
+    return arr[bounding_box]
+    """
+    Trims away the zero-filled slices that surround many 3/4D arrays
+
+    Parameters
+    ----------
+    ndarray: (S x T) array_like
+        an array with signal, surrounded by slices that contain only zeros
+        that should be removed.
+
+    Returnes
+    ---------
+    ndarray: (S x T) array_like
+        an array with reduced dimensions, such that the array contains only
+        non_zero values from edge to edge.
+    """
+
 
 
 def write_comp_figs(ts, mask, comptable, mmix, n_vols,
-                    acc, rej, midk, empty, ref_img):
+                    acc, rej, midk, empty, ref_img, out_dir,
```

Comment:
This function shouldn't need `acc`, `rej`, `midk`, or `empty` since it already gets `comptable`.

Reply:
Also we can infer `n_vols` from `ts` or `mmix`.

Reply:
I use each of those to color the lines, rather than redoing the work to pull it out of the comp table again. Unless those are going to be removed in the future?

Reply:
Good point. I'll fiddle with that in a bit.

Reply:
should work well now, else statement now will deal with a new classification somewhat gracefully by plotting a grey line (I think, testing now)